services/learn-card-network/brain-service/test/profiles.spec.ts

```diff
@@ -1698,4 +1492,112 @@ describe('Profiles', () => {
+                ).resolves.not.toThrow();
+            });
+
+            it('does not allow using the same challenge more than once', async () => {
+                try {
+                    // Generate the first invite with the challenge 'nice'
+                    await userA.clients.fullAuth.profile.generateInvite({ challenge: 'nice' });
+                    console.log('First invite generated with challenge: "nice"');
+
+                    // Try to generate another invite with the same challenge 'nice'
+                    await userA.clients.fullAuth.profile.generateInvite({ challenge: 'nice' });
+                } catch (error) {
+                    console.log('Caught error:', error); // Log the actual error object
+                    expect(error).toMatchObject({
+                        code: 'CONFLICT',
+                        message: 'Challenge already in use!',
+                    });
+                }
+            });
+
+
+
+
+            it('allows setting the expiration date', async () => {
+                const result = await userA.clients.fullAuth.profile.generateInvite({ expiration: '24h' });
+                expect(result.expiresIn).toBe(24 * 60 * 60);
+            });
+        });
+
+        describe('connectWithInvite', () => {
+            it('allows users to connect via invite challenge', async () => {
+                const { challenge } = await userB.clients.fullAuth.profile.generateInvite();
+
+                await expect(
+                    userA.clients.fullAuth.profile.connectWithInvite({ profileId: 'userb', challenge })
+                ).resolves.toBe(true);
+
+                const oneConnection = await userA.clients.fullAuth.profile.connections();
+
+                expect(oneConnection).toHaveLength(1);
+                expect(oneConnection[0]!.profileId).toEqual('userb');
+            });
+
+            it('does not allow users to connect with an expired challenge', async () => {
+                const { challenge } = await userB.clients.fullAuth.profile.generateInvite({
+                    expiration: '1s',
+                });
+
+                console.log(`Test - Generated challenge: ${challenge}`);
+
+                await new Promise(resolve => setTimeout(resolve, 2000));
+
+                try {
+                    const result = await userA.clients.fullAuth.profile.connectWithInvite({ profileId: 'userb', challenge });
+                    console.log(`Test - Unexpected success: ${result}`);
+                } catch (error) {
+                    console.log(`Test - Caught error: ${error.message}`);
+                    throw error;
+                }
+
+                await expect(
+                    userA.clients.fullAuth.profile.connectWithInvite({ profileId: 'userb', challenge })
+                ).rejects.toMatchObject({ code: 'NOT_FOUND', message: 'Invite not found or has expired' });
+            });
```

Comment:
This test appears broken. When running it, I see "Test - Unexpected success:" get logged, which implies that the link _did_ work after expiring, but if it _didn't_ work, then this should have thrown an error! Then the assertion at the bottom seems to be just trying to connect again and not working, which you would expect!